src/Slick/Pandoc.hs

```diff
@@ -50,76 +67,79 @@ unPandocM p = do
+--   build.
+makePandocReader' :: (FromJSON a) => PandocReader textType
+                  -> textType
+                  -> Action (Pandoc, a)
 makePandocReader' readerFunc text = do
   (pdoc, meta)  <- makePandocReader readerFunc text
   convertedMeta <- convert meta
   return (pdoc, convertedMeta)
 
+--------------------------------------------------------------------------------
+
+
 -- | Load in a source document using the given 'PandocReader', then render the 'Pandoc'
--- into text using the given 'PandocWriter'.
--- Returns a 'Value' wherein the rendered text is set to the "content" key and 
--- any metadata is set to its respective key in the 'Value'
-loadUsing :: PandocReader textType -> PandocWriter -> textType -> Action Value
+--   into text using the given 'PandocWriter'.
+--   Returns a 'Value' wherein the rendered text is set to the "content" key and
+--   any metadata is set to its respective key in the 'Value'
+loadUsing :: PandocReader textType
+          -> PandocWriter
+          -> textType
+          -> Action Value
 loadUsing reader writer text = do
   (pdoc, meta) <- makePandocReader reader text
   outText      <- unPandocM $ writer pdoc
   let withContent = meta & _Object . at "content" ?~ String outText
   return withContent
 
 -- | Like 'loadUsing' but allows also deserializes the 'Value' into any object
--- which implements 'FromJSON'.  Failure to deserialize will fail the Shake
--- build.
-loadUsing'
-  :: (FromJSON a)
-  => PandocReader textType
-  -> PandocWriter
-  -> textType
-  -> Action a
-loadUsing' reader writer text = loadUsing reader writer text >>= convert
-
--- | Attempt to convert between two JSON serializable objects (or 'Value's).
--- Failure to deserialize fails the Shake build.
-convert :: (FromJSON a, ToJSON a, FromJSON b) => a -> Action b
-convert a = case fromJSON (toJSON a) of
-  Success r   -> pure r
-  Error   err -> fail $ "json conversion error:" ++ err
-
--- | Flatten a Pandoc 'Meta' into a well-structured JSON object, rendering Pandoc
--- text objects into plain strings along the way.
-flattenMeta :: Meta -> Value
-flattenMeta (Meta meta) = toJSON $ fmap go meta
- where
-  go :: MetaValue -> Value
-  go (MetaMap     m) = toJSON $ fmap go m
-  go (MetaList    m) = toJSONList $ fmap go m
-  go (MetaBool    m) = toJSON m
-  go (MetaString  m) = toJSON m
-  go (MetaInlines m) = toJSON $ stringify m
-  go (MetaBlocks  m) = toJSON $ stringify m
+--   which implements 'FromJSON'.  Failure to deserialize will fail the Shake
+--   build.
+loadUsing' :: (FromJSON a)
+           => PandocReader textType
+           -> PandocWriter
+           -> textType
+           -> Action a
+loadUsing' reader writer text =
+  loadUsing reader writer text >>= convert
+
+-- | Generalized Entity Loader
+--
+loadEntity :: FromJSON b => EntityFilePath a
+           -> Action b
+loadEntity (EntityFilePath path) =
+  subloadEntity path
+
+subloadEntity :: FromJSON b => FilePath -> Action b
```

Comment:
Can you comment on what `subloadEntity` does?

Reply:
@mgajda description added

Reply:
comment added